addons/mrp/tests/test_warehouse_multistep_manufacturing.py

```diff
@@ -91,6 +91,46 @@ def test_02_warehouse_twostep_manufacturing(self):
         self.assertEqual(len(self.warehouse.pbm_route_id.rule_ids), 3)
         self.assertEqual(self.warehouse.manufacture_pull_id.location_dest_id.id, self.warehouse.lot_stock_id.id)
 
+    def test_manufacturing_2_steps_sublocation(self):
+        """Check having stock in child location of pre prod make 2 steps manufacturing"""
```

Comment:
I'd rewrite that, as it's not really what's checked in the test (or at least left me wonder for a few minutes :sweat_smile:).
More that it checks that a manufacture picking type pulling from a sublocation of pre-prod properly triggers the next rule.